combat-trainer.lic

```diff
@@ -199,8 +199,9 @@ class SetupProcess
     end
 
     # Exclude current skill so that a new one is selected.
-    new_weapon_skills = weapon_training.keys.reject { |skill| skill == game_state.weapon_skill }
-    new_weapon_skill = game_state.sort_by_rate_then_rank(new_weapon_skills, @priority_weapons).first
+    new_weapon_skill = weapon_training
+      .reject { |skill, _| skill == game_state.weapon_skill }
+      .min_by { |skill, _| [DRSkill.getxp(skill), @priority_weapons.include?(skill) ? -1 : 0, DRSkill.getrank(skill)] }.first
```

Comment:
⚡ Merge conflict, this is inadvertently reverting changes introduced in https://github.com/rpherbig/dr-scripts/pull/4686 that should be kept